[charlesliyifeng] iP

charlesliyifeng opened the pull request:
# Astra
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

> "I didn't say this" – Sun Tzu

Astra remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :+1: :+1: 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a _Java_ programmer, you can use it to practice _Java_ too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

charlesliyifeng committed:
level 0

charlesliyifeng committed:
add echo

charlesliyifeng committed:
Add add and list

charlesliyifeng committed:
Add mark + unmark
Create Task and TaskList

charlesliyifeng committed:
Fix format

charlesliyifeng committed:
Add Todo, Event, Deadline

charlesliyifeng committed:
Add support for different task types

charlesliyifeng committed:
Add automated testing

charlesliyifeng committed:
Add error handling

charlesliyifeng committed:
Add delete task

charlesliyifeng committed:
Add TaskType enum

charlesliyifeng committed:
Add Javadoc comments

charlesliyifeng committed:
Update gitignore

charlesliyifeng committed:
Add write tasks to file

charlesliyifeng committed:
Add read tasks from file

charlesliyifeng committed:
Merge branch 'branch-Level-7'

charlesliyifeng committed:
Update tasks to store dates using LocalDateTime

charlesliyifeng committed:
Merge branch 'branch-Level-8'

charlesliyifeng committed:
Add Ui class

charlesliyifeng committed:
Add Storage class

charlesliyifeng committed:
Add Parser and Commands

charlesliyifeng committed:
Merge branch 'more-OOP'

charlesliyifeng committed:
Organize Astra into packages

charlesliyifeng committed:
Merge branch 'add-gradle-support' of https://github.com/charlesliyifeng/ip

charlesliyifeng committed:
Add Gradle and checkstyle

charlesliyifeng committed:
Add JUnit test cases for Parser and AddCommand

charlesliyifeng committed:
Update JAR version

charlesliyifeng committed:
Add JavaDoc header comments

charlesliyifeng committed:
Tweak code to comply with coding standards

charlesliyifeng committed:
Add find command

charlesliyifeng committed:
Merge branch 'branch-A-JavaDoc'

charlesliyifeng committed:
Merge branch 'branch-A-CodingStandard'

charlesliyifeng committed:
Merge branch 'branch-Level-9'

charlesliyifeng committed:
Update README

charlesliyifeng committed:
Merge branch 'branch-A-CheckStyle'

charlesliyifeng committed:
Add GUI support

charlesliyifeng committed:
Merge branch 'branch-Level-10'

MinkhantEain reviewed (commented):
LGTM. Well done 🥳

lim-jt commented on src/main/java/astra/Storage.java:
```diff
@@ -0,0 +1,66 @@
+        ArrayList<Task> tasks = new ArrayList<>();
+        File f = new File(path + FILENAME);
+        try {
+            Scanner inp = new Scanner(f);
```
Perhaps instead of inp you could use a clearer name like inputScanner?

lim-jt commented on src/main/java/astra/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @return List of tasks in text file format.
+     */
+    public String toText() {
```
Perhaps this could be further differentiated from toString() to illustrate that the string is to be converted for the text file format, maybe toFileText()?

lim-jt commented on src/main/java/astra/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        this.inp = new Scanner(System.in);
+    }
+
+    private static String formatMsg(String msg) {
```
Perhaps a more descriptive name of what 'format' here means would be better?

lim-jt commented on src/main/java/astra/Ui.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Say goodbye to the user.
+     */
+    public void goodbye() {
```
Perhaps this could be named as a verb, maybe printGoodbye?

lim-jt reviewed (commented):
LGTM, just need to fix a few nits

charlesliyifeng committed:
change release version

charlesliyifeng committed:
Add assertions

charlesliyifeng committed:
Update method names to follow coding standard

charlesliyifeng committed:
TaskList class: Extract private methods

TaskList methods (e.g. delete, markAsDone) have common behaviors (e.g. index-1, catch IndexOutofBounds).

There are multiple levels of abstraction and unnecessary code duplication.

Extracting common behavior into private get and remove methods to maintain SLAP. Changing access modifier of get since it is only used inside the class.

Using private get and remove methods keeps the abstraction barrier because only they directly interact with tasks.

charlesliyifeng committed:
Improve readability of conditional statement

charlesliyifeng committed:
MainWindow class: Extract methods sendUserMessage and sendAstraMessage

MainWindow methods have some common behavior (e.g. sending messages)

The common behaviors across methods cause code duplication.

Extracting the common behavior into two private methods allows for code reuse.

charlesliyifeng committed:
Add streams

charlesliyifeng committed:
Merge pull request #1 from charlesliyifeng/branch-A-Assertions

Add assertions

charlesliyifeng committed:
Merge branch 'master' into branch-A-CodeQuality

charlesliyifeng committed:
Merge pull request #2 from charlesliyifeng/branch-A-CodeQuality

Branch a code quality

charlesliyifeng committed:
Merge branch 'master' into branch-A-Streams

charlesliyifeng committed:
Merge pull request #3 from charlesliyifeng/branch-A-Streams

Add streams

charlesliyifeng committed:
Add sort by deadline command

charlesliyifeng committed:
Merge branch 'branch-A-Sort'

charlesliyifeng committed:
Fix coding standard

charlesliyifeng committed:
Add title and icon

charlesliyifeng committed:
Add user guide

charlesliyifeng committed:
Update user guide description

charlesliyifeng committed:
Update README

charlesliyifeng committed:
Fix invalid task description bug

charlesliyifeng committed:
Add validity checks for event dates
